// value-looker/src/config.js
var colors = require('colors');

var version = '(v2.1.2)';

var colorTheme = {
    silly: 'rainbow',
    verbose: 'cyan',
    info: 'green',
    data: 'grey',
    help: 'cyan',
    warn: 'yellow',
    debug: 'blue',
    error: 'red'
};
colors.setTheme(colorTheme);

var helpText = '' +
    '/////////////////////////////////////////\n' +
    '欢迎使用Value-Looker\n'.info +
    '/////////////////////////\n\n' +
    '' +
    '【作用介绍】\n\n'.info +
    '' +
    '    精美查值器(Value-Looker)的主要功能是在控制台中以相应格式打印JavaScript的值（不论值的类型是什么都能打印）。'.data +
    '另外，它还有另一个重要的功能：可以获取格式化的对象或Json对象字符串。这个功能在个在使用node生成json文件时很有用！\n\n'.data +
    '' +
    '【快速开始】\n\n'.info +
    '' +
    '    1.安装：\n\n'.help +
    '        $ npm install --save value-looker\n\n'.data +
    '    2.引用：\n\n'.help +
    '        // demo.js\n'.data +
    '        var valueLooker = require("value-looker");\n'.data +
    '        valueLooker("help");\n\n'.data +
    '    3.运行：\n\n'.help +
    '        $ node demo.js\n\n'.data +
    '' +
    '【用法说明】\n\n'.info +
    '' +
    '    1.参数：\n\n' +
    '        valueLooker(value, config);\n\n'.data +
    '        @arg value [any | "help"] value为需要打印或转换的值，类型为any；当value为"help"时，则打印这个帮助信息\n'.data +
    '        @arg config [object(keys: title, theme, indent, json, val) optional] config是一个object类型的配置\n'.data +
    '        @arg config.title [string, optional] 看值器的标题\n'.data +
    '        @arg config.them [string, optional] 打印值的颜色主题\n'.data +
    '        @arg config.indent [number, optional] 返回的格式化的缩进值\n'.data +
    '        @arg config.json [boolean, optional] 是否返回Json化的值\n'.data +
    '        @arg config.val [boolean, optional] 如果是true，将直接返回得到格式化的对象或Json对象\n\n'.data +
    '' +
    '    2.colors theme：\n\n'.help +
    '        silly'.silly +
        '    verbose'.verbose +
        '    info'.info +
        '    data'.data +
        '    help'.help +
        '    warn'.warn +
        '    debug'.debug +
        '    error\n\n'.error +
    '' +
    '/////////////////////////\n' +
    'Thank You!\n'.info +
    '/////////////////////////////////////////' +
    '';

module.exports = {
    version: version,
    colorTheme: colorTheme,
    helpText: helpText
};
